bayesian/utils.py

```diff
@@ -410,87 +411,123 @@ class CveByDateEcosystemUtils:
+            self._cve_sources = cve_sources
+            self._date_range = date_range
 
     def get_cves_by_date_ecosystem(self):
         """Call graph and get CVEs by date and ecosystem."""
-        script = self.cve_nodes_by_date_ecosystem_script_template
-        self._ecosystem = self._ecosystem.lower()
-        bindings = {'modified_date': self._bydate, 'ecosystem': self._ecosystem}
-        return self.get_cves(script, bindings)
+        didx = pd.date_range(end=self._bydate, periods=self._date_range)
+        gremlin_json = []
 
-    def get_cves(self, script, bindings):
-        """Call Gremlin and get the CVE information."""
-        json_payload = self.prepare_payload(script, bindings)
-        response = post(gremlin_url, json=json_payload)
-        cve_list = self.prepare_response(response.json())
+        try:
+            for dt in didx:
+                dt = datetime.datetime.strftime(dt, '%Y%m%d')
+                # Gremlin script execution
+                script = self.cve_nodes_by_date_ecosystem_script_template_all \
+                    if self._cve_sources == 'all' \
+                    else self.cve_nodes_by_date_ecosystem_script_template
+
+                self._ecosystem = self._ecosystem.lower()
+                bindings = {
+                    'modified_date': dt, 'ecosystem': self._ecosystem,
+                    'cve_sources': self._cve_sources
+                }
+                """Call Gremlin and get the CVE information."""
+                json_payload = self.prepare_payload(script, bindings)
+
+                response = post(gremlin_url, json=json_payload)
+                gremlin_json.extend(response.json().get('result', {}).get('data', []))
+        except Exception as e:
+            return ' '.join([type(e), ':', str(e)])
+
+        cve_list = self.prepare_response_cve(gremlin_json)
         return cve_list
 
+    def get_cves_epv_by_date(self):
+        """Call graph and get CVEs by date and ecosystem."""
+        script = self.cve_epv_nodes_by_date_script_template
+        bindings = {'cve_id': self._cve_id}
+        """Call Gremlin and get the CVE EPV information."""
+        json_payload = self.prepare_payload(script, bindings)
+        try:
+            response = post(gremlin_url, json=json_payload)
+        except Exception as e:
+            return ' '.join([type(e), ':', str(e)])
+        epv_list = self.prepare_response_epv(response.json())
+        return epv_list
+
     def prepare_payload(self, script, bindings):
         """Prepare payload."""
         payload = {'gremlin': script, 'bindings': bindings}
 
         return payload
 
-    def prepare_response(self, gremlin_json):
+    def prepare_response_cve(self, gremlin_json):
         """Prepare response to be sent to user based on Gremlin data."""
         cve_list_add = []
-        cve_list_remove = []
+        # resp = gremlin_json.get('result', {}).get('data', [])
+        for cve in gremlin_json:
+            cve_dict = {
```

Comment:
You can add sources as well sent as the response

Reply:
nice suggestion. Will do that. thanx